client/blocks/inline-help/popover.jsx

```diff
@@ -384,7 +385,8 @@ function mapStateToProps( state, { moment } ) {
 
 	const isAllowedToUseClassic =
 		! isSiteUsingFullSiteEditing( state, siteId ) && ! isUsingGutenbergPageTemplates;
-	const showOptOut = optInEnabled && isGutenbergEditor && isAllowedToUseClassic;
+	const showOptOut =
+		isGutenbergOptOutEnabled( state, siteId ) && isGutenbergEditor && isAllowedToUseClassic;
```

Comment:
Minor, non-blocking: If we only have controls displayed in the block-editor we can maybe also move over the isGutenbergEditor and isAllowedToUseClassic logic into the selector as well.

Reply:
This is handled in https://github.com/Automattic/wp-calypso/pull/36464/files.

(This moves these checks to the server.)